### PR TITLE
apps/example/testcase/filesystem: Add tmpfs Mountpt operations testcase

### PR DESCRIPTION
**Pull request by KimDongEon**
Add a testcase to cover the code of tmpfs mountpt operations.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [7badf7fd68b2d151bbb3bf3811d8f115ad94b3c3] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7badf7fd68b2d151bbb3bf3811d8f115ad94b3c3] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ad0c131595f025f0f59b3a1a7a8e018e567f5572] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ad0c131595f025f0f59b3a1a7a8e018e567f5572] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	ret = write(fd1, write_buf, strlen(write_buf));
+	TC_ASSERT_EQ("write", ret, strlen(write_buf));
+
+	ret = lseek(fd1, 0, SEEK_SET);
+	TC_ASSERT_EQ("lseek", ret, OK);
+
+	memset(read_buf, 0, sizeof(read_buf));
+	ret = read(fd1, read_buf, sizeof(read_buf));
+	TC_ASSERT_GEQ("read", ret, 0);
+	TC_ASSERT_EQ("read", strcmp(read_buf, TMPFS_TEST_CONTENTS), 0);
```
Comment (yashwanth686007) [inline apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c]: Please perform cleanup if read/write fails.

File: apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+	ret = write(fd1, write_buf, strlen(write_buf));
+	TC_ASSERT_EQ("write", ret, strlen(write_buf));
+
+	ret = lseek(fd1, 0, SEEK_SET);
+	TC_ASSERT_EQ("lseek", ret, OK);
+
+	memset(read_buf, 0, sizeof(read_buf));
+	ret = read(fd1, read_buf, sizeof(read_buf));
+	TC_ASSERT_GEQ("read", ret, 0);
```
Comment (yashwanth686007) [inline apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c]: I think its better to compare the value of ret with buffer length.

File: apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	TC_ASSERT_EQ("read", strcmp(read_buf, TMPFS_TEST_CONTENTS), 0);
+
+	ret = ioctl(fd1, FIOC_MMAP, (unsigned long)&size);
+	TC_ASSERT_EQ("ioctl", ret, OK);
+
+	fd2 = dup(fd1);
+	TC_ASSERT_GEQ("dup", fd2, 0);
+
```
Comment (yashwanth686007) [inline apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c]: Please close fd2 also after using it.

File: apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c (hunk 2 above)
Comment (KimDongEon) [reply apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c]: ok, modify

File: apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c (hunk 3 above)
Comment (KimDongEon) [reply apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c]: I will close fd2

File: apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c (hunk 1 above)
Comment (KimDongEon) [reply apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c]: I will modify

Comment (KimDongEon) [reply apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c]: I added cleanup

File: apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c (hunk 2 above)
Comment (KimDongEon) [reply apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c]: Changed to compare length of read string.

File: apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c (hunk 3 above)
Comment (KimDongEon) [reply apps/examples/testcase/le_tc/filesystem/tc_fs_tmpfs.c]: I will close fd2

**Comment by seinfra** [comment]
Target : [b7996091772b73ba701a33910108e5229e0c8ed6] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b7996091772b73ba701a33910108e5229e0c8ed6] - Code Rule Check OK.

**Pull request merged**